### PR TITLE
*: upgrade go version to 1.16 and substitute ioutil

### LINKED ISSUE #3672: Upgrade Go version to 1.16
## Development Task

So far, PD development uses Go 1.13 which is end of life for a while. Also for supporting running on Apple M1 silicon, it is required to upgrade to 1.16.

@CadmusJiang and @zhouqiang-cl are in charge for upgrading Go version to 1.16 and most of validation works are done manually. Hopefully we get it done in this month.

This also supports TiDB stacks to upgrade Go version to 1.16 https://github.com/pingcap/tidb/issues/22198

### PR DESCRIPTION
**Pull request by tisonkun**
### What problem does this PR solve?

closes #3672 

### What is changed and how it works?

- Update Go version from 1.13 to 1.16
- Substitute deprecated ioutil based on https://golang.org/doc/go1.16#ioutil

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by tisonkun** [comment]
```
2021-05-27T06:27:03.6677937Z coverage: 11.8% of statements in ./...
2021-05-27T06:27:03.6678539Z goleak: Errors on successful test run: found unexpected goroutines:
2021-05-27T06:27:03.6679582Z [Goroutine 2800 in state sync.Cond.Wait, with sync.runtime_notifyListWait on top of the stack:
2021-05-27T06:27:03.6680834Z goroutine 2800 [sync.Cond.Wait]:
2021-05-27T06:27:03.6681660Z sync.runtime_notifyListWait(0xc000ea8b90, 0x0)
2021-05-27T06:27:03.6682487Z 	/opt/hostedtoolcache/go/1.16.4/x64/src/runtime/sema.go:513 +0xf8
2021-05-27T06:27:03.6683097Z sync.(*Cond).Wait(0xc000ea8b80)
2021-05-27T06:27:03.6684155Z 	/opt/hostedtoolcache/go/1.16.4/x64/src/sync/cond.go:56 +0x90
2021-05-27T06:27:03.6684941Z net/http.(*connReader).abortPendingRead(0xc000b85da0)
2021-05-27T06:27:03.6685624Z 	/opt/hostedtoolcache/go/1.16.4/x64/src/net/http/server.go:740 +0x1a5
2021-05-27T06:27:03.6686221Z net/http.(*response).finishRequest(0xc0043f9ce0)
2021-05-27T06:27:03.6686984Z 	/opt/hostedtoolcache/go/1.16.4/x64/src/net/http/server.go:1633 +0x145
2021-05-27T06:27:03.6687579Z net/http.(*conn).serve(0xc000b501e0, 0x41b2478, 0xc000ea8b40)
2021-05-27T06:27:03.6688190Z 	/opt/hostedtoolcache/go/1.16.4/x64/src/net/http/server.go:1957 +0x8d7
2021-05-27T06:27:03.6688749Z created by net/http.(*Server).Serve
2021-05-27T06:27:03.6689339Z 	/opt/hostedtoolcache/go/1.16.4/x64/src/net/http/server.go:3013 +0x645
2021-05-27T06:27:03.6689694Z 
2021-05-27T06:27:03.6690274Z  Goroutine 2747 in state sync.Cond.Wait, with sync.runtime_notifyListWait on top of the stack:
2021-05-27T06:27:03.6691369Z goroutine 2747 [sync.Cond.Wait]:
2021-05-27T06:27:03.6692083Z sync.runtime_notifyListWait(0xc000dc6110, 0x0)
2021-05-27T06:27:03.6692768Z 	/opt/hostedtoolcache/go/1.16.4/x64/src/runtime/sema.go:513 +0xf8
2021-05-27T06:27:03.6693854Z sync.(*Cond).Wait(0xc000dc6100)
2021-05-27T06:27:03.6694371Z 	/opt/hostedtoolcache/go/1.16.4/x64/src/sync/cond.go:56 +0x90
2021-05-27T06:27:03.6694993Z net/http.(*connReader).abortPendingRead(0xc000ea7590)
2021-05-27T06:27:03.6695669Z 	/opt/hostedtoolcache/go/1.16.4/x64/src/net/http/server.go:740 +0x1a5
2021-05-27T06:27:03.6696268Z net/http.(*response).finishRequest(0xc000dee460)
2021-05-27T06:27:03.6697004Z 	/opt/hostedtoolcache/go/1.16.4/x64/src/net/http/server.go:1633 +0x145
2021-05-27T06:27:03.6697950Z net/http.(*conn).serve(0xc007667360, 0x41b2478, 0xc000dc60c0)
2021-05-27T06:27:03.6698550Z 	/opt/hostedtoolcache/go/1.16.4/x64/src/net/http/server.go:1957 +0x8d7
2021-05-27T06:27:03.6699099Z created by net/http.(*Server).Serve
2021-05-27T06:27:03.6699875Z 	/opt/hostedtoolcache/go/1.16.4/x64/src/net/http/server.go:3013 +0x645
2021-05-27T06:27:03.6700630Z 
2021-05-27T06:27:03.6701289Z ]
2021-05-27T06:27:03.6701903Z FAIL	github.com/tikv/pd/tests/server/api	19.924s
2021-05-27T06:27:03.6702332Z FAIL
2021-05-27T06:27:03.6703099Z make: *** [Makefile:161: test-with-cover] Error 1
2021-05-27T06:27:03.6709005Z ##[error]Process completed with exit code 2.
```

I have no idea why goleak failed. I don't think it is related to this pr.

File: pkg/testutil/leak.go [hunk 1]
```diff
@@ -28,6 +28,8 @@ var LeakOptions = []goleak.Option{
 	goleak.IgnoreTopFunction("google.golang.org/grpc.(*addrConn).resetTransport"),
 	goleak.IgnoreTopFunction("google.golang.org/grpc.(*Server).handleRawConn"),
 	goleak.IgnoreTopFunction("go.etcd.io/etcd/pkg/logutil.(*MergeLogger).outputLoop"),
+	// FIXME: Go 1.16 changes call stack, I have no idea where this comes from.
```
Comment (disksing) [inline pkg/testutil/leak.go:31]: Can remove the comment. For most of the contents of the list, we don't know where they came from.

File: server/api/status_test.go [hunk 2]
```diff
@@ -15,10 +15,9 @@ package api
 
 import (
 	"encoding/json"
-	"io/ioutil"
-
 	. "github.com/pingcap/check"
 	"github.com/tikv/pd/server/versioninfo"
+	"io"
```
Comment (disksing) [inline server/api/status_test.go:20]: Seems not in good order.

File: pkg/testutil/leak.go (hunk 1 above)
Comment (tisonkun) [reply pkg/testutil/leak.go:31]: fine

File: server/api/status_test.go (hunk 2 above)
Comment (tisonkun) [reply server/api/status_test.go:20]: will update

**Review by disksing — APPROVED** [review]

**Review by nolouch — APPROVED** [review]
lgtm

**Pull request merged**